src/main/java/MyExecutor.java

```diff
@@ -0,0 +1,39 @@
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import org.apache.commons.collections4.ListUtils;
+
+public class MyExecutor {
+    public static final int THREADS = Runtime.getRuntime().availableProcessors();
+    private final List<Integer> numbers;
+
+    public MyExecutor(List<Integer> numbers) {
+        this.numbers = numbers;
+    }
+
+    public Integer calculate() {
+        ExecutorService executorService = Executors.newFixedThreadPool(THREADS);
+        List<List<Integer>> partition = ListUtils.partition(numbers, numbers.size() / THREADS);
+        List<MyCallable> callables = new ArrayList<>();
```

Comment:
```suggestion
        List<Callable> tasks = new ArrayList<>();
```
use abstraction where possible